misk-hibernate-testing/src/main/kotlin/misk/jdbc/VitessScaleSafetyChecks.kt

```diff
@@ -103,6 +106,8 @@ class VitessScaleSafetyChecks(
   }
 
   inner class TableScanDetector : ExtendedQueryExecutionListener() {
+    private val tablePattern = Maps.newConcurrentMap<String, Pattern>()
```

Comment:
For what it’s worth, I’ve been encouraging lighter-weight datastructures instead of ConcurrentHashMap
https://github.com/cashapp/misk/pull/1477#discussion_r413028978

Reply:
In this case I suspect you could avoid caching altogether and get good enough performance

Reply:
Not sure about no need for caching :-) This runs on EVERY query so compiling a regexp pattern might get really expensive...

I don't think we need a full blown CHM though, it's not very concurrent but does SHM have the very convenient `computeIfAbsent` method?

Reply:
makes sense!. Changed to ```Collections.synchronizedMap(...)```